src/Hosting/TestHost/src/HttpResetTestException.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.TestHost
+{
+    /// <summary>
+    /// Used to surface to the test client that the application invoked <see cref="IHttpResetFeature.Reset"/>
+    /// </summary>
+    public class HttpResetTestException : Exception
```

Comment:
Is throwing an exception for reset and abort the best approach? In Kestrel the response will be ended and the body immediately returned to the user. That's quite different from an exception.

Reply:
In Kestrel Reset fully aborts the request and response, the same as Abort. You're describing CompleteAsync, not Reset. 

A real client would also throw in the Reset/Abort scenario.

Reply:
What should happen if you first call CompleteAsync and then Reset/Abort is a more interesting topic. I'll figure that out when I implement CompleteAsync.

Reply:
> In Kestrel Reset fully aborts the request and response, the same as Abort. You're describing CompleteAsync, not Reset.

I was thinking about content that was flushed before abort/reset. Yes you're right that they raise an error in HttpClient.

> What should happen if you first call CompleteAsync and then Reset/Abort is a more interesting topic. I'll figure that out when I implement CompleteAsync.

no-op?

Reply:
It's not a no-op for a real client that is still uploading. And you have a scenario for that ordering gRPC. I think I'll add a new feature just for the client to look at so it can have a Task to await for the request pipeline to unwind and observe any late exceptions.

Reply:
I would have thought that complete would already have closed the request body. You know a lot more about this area than me so I'll stop guessing 😄

Reply:
Yeah, CompleteAsync only affects the response, the request body stays open until you unwind the app or the client closes it. Reset or Abort will kill it though.